lib/aarch64/parser.rb

```diff
@@ -218,10 +611,20 @@ def _next_token
         end
       elsif str = @scan.scan(SYS_REG_SCAN)
         [:SYSTEMREG, SYS_REG_MAP[str.downcase]]
-      elsif str = @scan.scan(/\w+/)
+      elsif str = @scan.scan(KEYWORDS_SCAN)
```

Comment:
Falling through to parsing `/\w+` as keywords wasn't working for the added requirement of parsing labels; I swapped to following most of the patterns in the 'very basic lexer' example [here](https://tenderlovemaking.com/2023/09/02/fast-tokenizers-with-stringscanner.html) as the first step to getting the tests passing :+1: